libs/filabridge/src/Variant.cpp

```diff
@@ -161,9 +164,9 @@ constexpr inline size_t fragment_variant_count() noexcept {
 }
 
 static_assert(reserved_is_not_valid());
-static_assert(reserved_variant_count() == 80);
-static_assert(valid_variant_count() == 48);
-static_assert(vertex_variant_count() == 16 - (2 + 0) + 4 - 0);        // 18
+static_assert(reserved_variant_count() == 160);
+static_assert(valid_variant_count() == 96);
+static_assert(vertex_variant_count() == 32 - (4 + 0) + 8 - 0);        // 36
```

Comment:
I think something might be wrong here. Only the number of vertex variants should have doubled. The total number of variants shouldn't be double because STE doesn't affect fragment variant count.
`filterVariantFragment` must be updated; STE is similar to SKN in that way.

Reply:
All of this variant logic is very confusing to me, I don't understand where a lot of these numbers come from or how they relate to the ASCII chart in Variant.h.

For instance, why is `valid_variant_count()` 48, but `vertex_variant_count() + fragment_variant_count()` only 42?

It seems like `filterVariantFragment` should _not_ be updated, because it's `&`-ing with bits that _should_ affect the fragment shader.

Reply:
Here's what the code does currently, which seems correct to me:

**No stereo**, 42 total variants:
- 18 vertex (4 DEP)
- 24 fragment (3 DEP)

**With stereo**, 60 total variants:
- 36 vertex (8 DEP)
- 24 fragment (3 DEP)

Reply:
It makes sense that valid variant count doubled (which, as we discussed, is not to be confused with the total number of generated shaders), because STE is completely orthogonal to any of the other variant bits.